### DISCUSSION TITLE
Orbit FVM Events-in-a-box

### DISCUSSION THREAD

**Discussion by trruckerfling** [filecoin-project/community#588] [General]
### Orbit FVM Event Preparation
[Resource folder](https://drive.google.com/drive/u/0/folders/1VvUrEmkzO2WUu2BJMaFI2fYMzK8kkoiA)

### Objective
Prepare Orbit members to deliver FVM events, in-person, to their local communities. These local community activations are part of the wider FEVM main net launch initiative, which will launch in early March.

<h2 style="font-family: Arial, sans-serif; margin: 30px 0px 10px; line-height: var(--line-height-medium); font-size: var(--font-up-2-rem); color: rgb(34, 34, 34); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Important dates</h2><div class="md-table" style="overflow-y: auto; margin: 1em 0px; color: rgb(34, 34, 34); font-family: Arial, sans-serif; font-size: 15.008px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

14 Jan - 28 Feb | Orbit FVM community events
-- | --
16 Jan | Hyperspace testnet goes live
18 Jan | Space Warp Hackathon Resource Kit ready
20 Jan - 10 Feb | Space Warp Hackathon
24 Feb - 5 Mar | ETHDenver
14 March | FEVM (FVM with Ethereum compatibility) main net launch
Mid Year | FVM native smart contracts launch

### Event Formats
There are 2 formats of Orbit FVM Events:

#### 1. FVM Meetup - Introduce your community to FVM, FEVM and have them try it out

- Hour 1: Introduction to FVM presentation, given by you. We will train you and give you the slides.
- Hours 2 & 3: Coding Dojo with an FVM Developer Advocate
- Expected expenses:
  - Venue for 3 hours
  - One meal served or happy hour
  - A/V Support and/or Photographer / Videographer
  - Some Orbit volunteers may need travel / accommodation covered
  - Swag (we will provide the design kit so that you just need to have the swag made in your locality)
- Expected cost: under $1,000 (depending on location)

#### 2. FVM Open Hack Day - Bring your community of FVM projects together to build

- We will give you the materials for workshops.
- The FVM team will provide bounties and other challenges for the participants.
- Expected expenses:
  - Hacker Venue for ~8 - 12 Hours
  - Lunch
  - Happy Hour / Dinner Served
  - A/V Support and/or Photographer / Videographer
  - Some Orbit volunteers may need travel / accommodation covered
  - Swag (we will provide the design kit so that you just need to have the swag made in your locality)
- Expected cost: Under $3,000 (depending on location)

#### Agenda
Timelapse (HH:MM) | Agenda
-- | --
00:00 - 00:05 | Host introduction and welcome
00:05 - 00:30 | Introducing FVM
00:30 - 01:00 | FVM Use Cases
01:00 - 01:30 | FVM with Ethereum Compatibility (FEVM)
01:30 - 02:00 | FEVM Hardhat Simple Storage Tutorial
02:00 - 02:30 | FEVM Remix Simplecoin Tutorial
02:30 - 03:00 | Food & Drinks

#### Presentation Resources 
Item | Resources
-- | --
Introducing FVM (25 mins) | [[UPDATED] Presentation deck](https://docs.google.com/presentation/d/1PHegl06OREoBsL06oz-iP7ugGaI6dpV6fuZIAVFXyQM/edit?usp=sharing), [[UPDATED] Video](https://www.youtube.com/watch?v=9m3oSxtAivs)
FVM Use Cases (30 mins) | [Slide 3 - 17](https://docs.google.com/presentation/d/1M4WBRSGpF8FhO6-uz2tvfLfFNFW2iHOJ/edit?usp=sharing&ouid=105045306321132519063&rtpof=true&sd=true), [Video](https://www.youtube.com/watch?v=EqLlyJ13DLY) (00:00 - 08:00)
FVM with Ethereum Compatibility (FEVM) (30 mins) | [Slide](https://docs.google.com/presentation/d/1QdUzq00sWtE4_DOS359KYJWQS4Wpy7XeC6P2lYWxaLI/edit?usp=sharing), [Video](https://www.youtube.com/watch?v=9m3oSxtAivs) 
FEVM Hardhat Tutorial (Minting ERC-20 token) 1 | [Video](https://www.youtube.com/watch?v=FJK2RDFtSUw)
FEVM Remix Tutorial (Minting ERC-20 token) | [Video](https://drive.google.com/file/d/194oRn0eXX1_lU3FE3QDc963-MEMw4j_J/view?usp=share_link), [Tutorial](https://docs.filecoin.io/developers/smart-contracts/how-tos/deploy-a-contract/)

#### Event Write-up (for marketing)
The FVM Open Hack Day gathers Filecoin, FVM and web3 developers to connect and create their first storage deal on Filecoin with a FVM smart contract. Following that, useful tooling and inspirational solution ideas will be shared and discussed, to help devs hack cool ideas on FVM, for a chance to win some bounties!

#### Event Prerequisites
Please advise registered attendees to prepare beforehand, to get the most out of the hack day:
- Attended the previous FVM Meetup AND/OR has a conceptual understanding of what FVM and FEVM can do (if you missed the FVM meetup, you can watch this [video](https://www.youtube.com/watch?v=9m3oSxtAivs))
- Comfortable with basic Solidity. You can watch this recommended [starter tutorial from Chainlink](https://www.youtube.com/watch?v=M576WGiDBdQ).
- Recommended to have a [basic understanding of Filecoin’s storage network](https://docs.filecoin.io/developers/introduction/overview/), as FVM builds programmability around it
- Recommended to check out [Request for Startups](https://rfs.fvm.dev/) ideas and [FEVM Early Builders](https://ecosystem.filecoin.io/?filters=enabled&tags=FVM) to start thinking about what you will build at the event!

#### Agenda
Timelapse (HH:MM) | Agenda
-- | --
00:00 - 00:05 | Host introduction and welcome
00:05 - 00:30 | Understand Filecoin storage deal making flow
00:30 - 01:00 | Set up your own Lotus node
01:00 - 02:00 | Break & Open Hack (mentorship provided)
02:00 - 03:00 | Create a Filecoin storage deal with FEVM actors (smart contracts)
03:00 - 04:00 | Build a smart DataDAO with FEVM
04:00 - 04:30 | FEVM tooling
04:30 - 06:30 | Break & Open Hack (mentorship provided)
06:30 - 07:30/08:00 (organizer can choose) | Open mic to showcase hack projects & #shipped party!

#### Presentation Resources & Speaker Notes
Item | Resources
-- | --
Understand Filecoin storage deal making flow | [Video](https://strn.pl/ipfs/bafybeiadse3iuucjjqslfwjzdcp7gfhe3sbsppzo5lotxfstgjhsr5uzqa?filename=filecoin_storage_deal_flow_talk.mp4), [Docs](https://github.com/filecoin-project/community/discussions/601) 
Set up your own Lotus node | [Video](https://bafybeifm5oa7qkvucvrvibavbcalo7sbuquocu5ewurcreizjil6uozrje.ipfs.w3s.link/), [Docs](https://docs.filecoin.io/developers/infrastructure/how-tos/spin-up-a-lite-node/)
Store data with FEVM and interact with the Filecoin storage deal network | [Video](https://www.youtube.com/watch?v=KcZbXSoUxz4), [Quickstart Docs (covered in 1/3 of video)](https://docs.filecoin.io/fvm/how-tos/quickstart/), [Hardhat Starter Kit](https://github.com/filecoin-project/FEVM-Hardhat-Kit)
DataDAO template with FEVM | [Video](https://drive.google.com/file/d/1F-V6R6BBXQA9y4nfde2Tvb-aepZfHewW/view?usp=share_link), [FEVM DataDAO Starter Kit](https://github.com/rk-rishikesh/DataDAO) 
FEVM tooling | [Share the cheat sheet resource for additional resources](https://github.com/filecoin-project/community/discussions/585)

### Additional Resources
[FVM Cheat Sheet](https://github.com/filecoin-project/community/discussions/585) (contains links to docs below, is a good guide to FVM content)
[Design assets](https://drive.google.com/drive/u/0/folders/1E6k2oO_on7JgvAgOUYqfXqZ5i4OaVeJZ)

**Comment by TimGuo7** [comment]
Awesome! This is super helpful to organize FVM related events.

**Comment by longfeiWan9** [comment]
I have participated in the FVM Orbit meetup in Shanghai and have noticed some outdated slides, as well as duplicate content which is not very good for a 2-3 hour short meetup. Here are a couple of my observations and recommendations.

- Introducing FVM (25 mins) - [slides](https://docs.google.com/presentation/d/1PHegl06OREoBsL06oz-iP7ugGaI6dpV6fuZIAVFXyQM/edit#slide=id.g1f8c6f38b5f_0_933): the content is heavily focus on EVM-compatibility which is actually better for 3th agenda (FVM with Ethereum Compatibility ). We should use a more general introduction to FVM. 
- FVM Use Cases (30 mins) - slides are outdated for the timeline. Also suggested using slide 1-17 because slides 18-26 is already duplicated with the Intro of FVM. If they play the video during the event, the content is duplicated. 
- FVM with Ethereum Compatibility - the video content include FVM intro, architecture, etc which is already covered in previous steps. If the organizer plays the video during the event, the audience will see the contents multiple times. 
- Attached video - I assume we do not want the event organizer to play video during the events,  right? 
   - If yes, there are many content duplications. 
   - if not, Maybe we should highlight that video is for the organizer to learn, not to play during the event.

**Comment by ProfessorMD1** [comment]
It's good to make an organised plan to guide people. I expect improvements and summary of everything. I don't know if I can be of help in that regard.

**Comment by Pfed-prog** [comment]
Sharing the materials from my talk on fvm https://x.com/Web3Family_/status/1705272010295709976?s=20
script: https://docs.google.com/document/d/1-tTKxJ8rCyMh8gRUCStHDn6fS6iMYvNaIMc9LTuSI6Y/edit?usp=sharing
slides: https://docs.google.com/presentation/d/1jXHcb0CNzzWbYy3YxvMf2yhV-4qMr2B9yE9BFhYjXZw/edit?usp=sharing

**Comment by excellence2705** [comment]
Great, this is helpful

**Comment by purity23** [comment]
wow, lots of resources, work made easy

**Comment by Pfed-prog** [comment]
I have further created blog posts

Would be great to get more feedback, the code is open-source, please, be free to make a pull request

https://dspyt.com/Filecoin-architecture
https://dspyt.com/Filecoin-FVM
